src/v2.rs

```diff
@@ -169,12 +173,101 @@ impl EszipV2Modules {
   }
 }
 
+#[derive(Debug, Clone, Copy)]
+struct Options {
+  /// Hash Function used to checksum the contents of the eszip when encoding/decoding
+  ///
+  /// If the eszip does not include the option, it defaults to `[Checksum::NoChecksum]` in >=v2.2
+  /// and `[Checksum::Sha256]` in older versions.  It is `None` when the eszip header includes a
+  /// checksum that this version of the library does not know.
+  checksum: Option<Checksum>,
+
+  /// Size in Bytes of the hash function digest.
+  ///
+  /// Defaults to the known length of the configured hash function. Useful in order to ensure forwards compatibility,
+  /// otherwise the parser does not know how many bytes to read.
+  checksum_size: Option<u8>,
+}
+
+impl Options {
+  fn default_for_version(magic: &[u8; 8]) -> Self {
+    let mut defaults = Self {
+      checksum: Some(Checksum::NoChecksum),
+      checksum_size: Default::default(),
+    };
+    if let ESZIP_V2_MAGIC | ESZIP_V2_1_MAGIC = magic {
+      // versions prior to v2.2 default to checksuming with SHA256
+      defaults.checksum = Some(Checksum::Sha256);
+    }
+    defaults
+  }
+}
+
+impl Default for Options {
+  fn default() -> Self {
+    Self::default_for_version(LATEST_VERSION)
+  }
+}
+
+impl Options {
+  /// Get the size in Bytes of the source hashes
+  ///
+  /// If the eszip has an explicit digest size, returns that. Otherwise, returns
+  /// the default digest size of the [`Self::checksum`]. If the eszip
+  /// does not have either, returns `None`.
+  fn checksum_size(self) -> Option<u8> {
+    self
+      .checksum_size
+      .or_else(|| Some(self.checksum?.digest_size()))
+  }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum Checksum {
+  NoChecksum = 0,
```

Comment:
Also, thanks for adding this! It should speed up running code using `deno compile`.